src/opendr/perception/object_detection_2d/yolov5/yolov5_learner.py

```diff
@@ -28,6 +33,13 @@ class YOLOv5DetectorLearner(Learner):
 
     def __init__(self, model_name, path=None, device='cuda', temp_path='.', force_reload=False):
         super(YOLOv5DetectorLearner, self).__init__(device=device, temp_path=temp_path)
+        self.device = device
+        self.model_directory = temp_path if path is None else path
+        self.model_name = model_name
+
+        if model_name not in self.available_models:
+             self.download(path='./', mode="pretrained", verbose=True, model_name=model_name)
+
         if model_name not in self.available_models:
```

Comment:
This will trigger when we have already downloaded a pretrained model. It is better to add a model name like 'yolov5s_trucks' and handle the download when this model name is provided. Otherwise, implicitly downloading a finetuned model instead of a generic one is not the behavior that a user will expect.

Reply:
I make all necessary changes, regarding this issue. So,  the PR is ready for review